### PR TITLE
 Add rule identifiers/references to graphql schema

### PR DESCRIPTION
**Pull request by akofink**
Built on top of https://github.com/RedHatInsights/compliance-backend/pull/171

Towards https://redhatmbu.aha.io/epics/COMP-E-99

### CODE REVIEW

**Comment by akofink** [comment]
This should be good for review now.

File: app/graphql/types/profile.rb [hunk 1]
```diff
@@ -11,7 +11,27 @@ class Profile < Types::BaseObject
...
+    field :rules, [::Types::Rule], null: true do
+      argument :identifier, String,
+               'Rule identifier to filter by', required: false
+      argument :references, [String],
+               'Rule references to filter by', required: false
+    end
+    def rules(args = {})
+      rules = object.rules
+
+      if args.dig(:identifier)
```
Comment (dLobatog) [inline app/graphql/types/profile.rb]: Seems like these 2 could be moved to a helper method? It's nearly identical code

Comment (dLobatog) [reply app/graphql/types/profile.rb]: Also it's separate to the rest of the methods down below

File: app/graphql/types/profile.rb [hunk 2]
```diff
@@ -11,7 +11,22 @@ class Profile < Types::BaseObject
...
+               'Rule identifier to filter by', required: false
+      argument :references, [String],
+               'Rule references to filter by', required: false
+    end
+
+    def rules(args = {})
+      rules = object.rules
+
+      rules = rules.with_identifier(args[:identifier]) if args.dig(:identifier)
+      rules = rules.with_references(args[:references]) if args.dig(:references)
```
Comment (akofink) [inline app/graphql/types/profile.rb]: How about this, @dLobatog?

**Pull request merged**